services/apps/src/apps/service/apps/jobs/sharing.clj

```diff
@@ -173,16 +173,28 @@
    (catch ce/clj-http-error? {:keys [body]}
      (str "unable to unshare result folder: " (:error_code (service/parse-json body))))))
 
-;; The apps client isn't used at this time, but it will be once we extend analysis sharing
-;; to HPC apps.
+(defn- unshare-input-file
+  [sharer sharee path]
+  (try+
+   (data-info/unshare-path sharer path sharee)
```

Comment:
I'm not completely sure how this would play out so I'm commenting to ensure you've thought about it: what happens if an input file is a file which the analysis sharee had access to before the analysis was shared? I think this is especially important in the case where the sharee might even be the input file's owner -- say, a professor has some data which they share with a student, who runs an analysis on that data and then shares the analysis back to the professor, or, in the case where the input file is a public file.

I guess that this call will fail if the sharer isn't the owner of the input file, which will cover most of the cases but might result in the unsharing failing, when it probably should continue in that case.

Thoughts?